currency/pair.go

```diff
@@ -150,11 +140,22 @@ func MatchPairsWithNoDelimiter(currencyPair string, pairs Pairs, pairFmt PairFor
 
 // GetFormatting returns the formatting style of a pair
 func (p Pair) GetFormatting() (PairFormat, error) {
-	if p.Base.UpperCase != p.Quote.UpperCase {
-		return PairFormat{}, fmt.Errorf("%w casing mismatch", errPairFormattingInconsistent)
+	if p.Base.isCaseSensitive() && p.Quote.isCaseSensitive() && (p.Base.upperCase != p.Quote.upperCase) {
+		return EMPTYFORMAT, fmt.Errorf("%w casing mismatch", errPairFormattingInconsistent)
+	}
+	upper := p.Base.isCaseSensitive() && p.Base.upperCase || p.Quote.isCaseSensitive() && p.Quote.upperCase
+	return PairFormat{Uppercase: upper, Delimiter: p.Delimiter}, nil
+}
+
+func (p Pair) hasFormatDifference(pairFmt PairFormat) bool {
+	return p.Delimiter != pairFmt.Delimiter ||
+		(p.Base.isCaseSensitive() && p.Base.upperCase != pairFmt.Uppercase) ||
+		(p.Quote.isCaseSensitive() && p.Quote.upperCase != pairFmt.Uppercase)
+}
+
+func (c Code) isCaseSensitive() bool {
+	if c.Item == nil {
+		return false
```

Comment:
This seems left over?
I felt like the original check should be okay here, personally. We just trust the `upperCase` field, and error if they're inconsistent.

Reply:
It's mostly to do with just numbers for either a base or quote, where the upperCase field is irrelevant, so for things like "ETH-420" or "eth-420" this bypasses the check required. Unless you have another way to do things or you want me to mutate the pair on creation so that it always conforms to string formatting but that seems like a weird step or maybe I am just not understanding. <-- high probability. 😆

Reply:
Actually, no, I think I'm good with this.
My thought was that if we don't *know* the casing, we can't derive the Formatting from it, therefore it's an error.
However I think your solution is better.
Just to check though:
When *might* we want to derive formatting from an ambiguously cased pair like `BTC-420`
If we don't have a current use-case, I'd vote for going for the simplest solution and erroring, removing the extra func for `isCaseSensitive` and leaving a comment saying that the option is there.
I normally push stuff like that to `mothball/feature_pair_getformat_ambiguous` to retrieve later.

So, up to you.

Reply:
defer to @gloriousCode 😆

Reply:
Resolving this:
I encountered this failure throughout ExchangeWrapperStandards, so it's neccessary. Happy with your change.